backend/Models/BaseModel.cs

```diff
@@ -1,15 +1,34 @@
 using System;
+using System.Diagnostics.CodeAnalysis;
 
 namespace Pims.Api.Models
 {
-    public abstract class BaseModel
+    public abstract class BaseModel : IEquatable<BaseModel>
     {
         #region Properties
         public DateTime CreatedOn { get; set; }
 
         public DateTime? UpdatedOn { get; set; }
 
         public string RowVersion { get; set; }
+
+        public override bool Equals(object obj)
+        {
+            return Equals(obj as BaseModel);
+        }
+
+        public bool Equals([AllowNull] BaseModel other)
+        {
+            return other != null &&
```

Comment:
I'm not really fussed about how we implement the actual comparison within this function as long as we override the IEquatable interface on objects like this so that we can do common array/list operations and also for convenience during testing. If this were Java I'd have concerns about the performance impact of using reflection for comparison on objects that can be in large collections, but I'm not sure if that is an issue in C#

Once again, let me note that the above is generated by visual studio, so while it may seem like a lot of work to add in all of the fields, none of that work is being done by hand.

Reply:
Also, the above isn't an entity and is not the base class of Agency.cs not sure if this class is referring to when you said: "see my comments on the base class".

Reply:
It sounds like you have something specific in mind with the reflection piece?

I found this as a possible solution, although I'm a bit wary of it because I'm not familiar with c# reflection syntax:

https://stackoverflow.com/questions/986572/how-to-quickly-check-if-two-data-transfer-objects-have-equal-properties-in-c#986617

For the above to work we would still need to add an equals override to all of the model classes, as the above does not work recursively on properties that are themselves reference types.

Reply:
Did you have something in mind for the reflection equality comparison? I found the below but I'm wary of it because I'm not familiar with the syntax.

https://stackoverflow.com/questions/986572/how-to-quickly-check-if-two-data-transfer-objects-have-equal-properties-in-c#986617

We would still need to include a call to the above in each overriden equals method in all of the models, as the above does not operate recursively on class properties that are reference types.

Reply:
We'll add this to TechDebt and create functions to compare two objects with reflection in the future.